### PR TITLE
Tos checkbox

### PR DESCRIPTION
**Pull request by cancng**

### CODE REVIEW

**Comment by cancng** [comment]
Preview for the change

CleanShot 2023-05-20 at 18 36 39

File: apps/reflect/src/components/auth/AuthContainer.tsx [hunk 1]
```diff
@@ -1,55 +1,110 @@
...
+} from "@mui/material";
 import { SiDiscord, SiTwitch } from "react-icons/si";
 import { signIn } from "next-auth/react";
 import Link from "next/link";
 import AuthLoginButton from "./AuthLoginButton";
+import { ChangeEvent, useState } from "react";
 
 const AuthContainer = () => {
+  const [acceptTos, setAcceptTos] = useState(false);
+  const [showNotAcceptedWarning, setShowNotAcceptedWarning] = useState(false);
```
Comment (mertsabinov) [inline apps/reflect/src/components/auth/AuthContainer.tsx:17]: Hello, thank you very much for your contribution. Could you please add the "useState" types to the lines in this section?

File: apps/reflect/src/components/auth/AuthContainer.tsx [hunk 2]
```diff
@@ -1,55 +1,110 @@
...
+        onClose={handleWarningClose}
+        anchorOrigin={{ horizontal: "center", vertical: "bottom" }}>
+        <Alert
+          variant="filled"
+          onClose={handleWarningClose}
+          severity="error"
+          sx={{ width: "100%" }}>
+          You must accept policies first!
+        </Alert>
+      </Snackbar>
```
Comment (mertsabinov) [inline apps/reflect/src/components/auth/AuthContainer.tsx:48]: please use the ready component included in the project in this section. The file name is `AppSnackbar.tsx`.

**Review by mertsabinov — CHANGES_REQUESTED** [review]
Hello, thank you very much for your contribution.

File: apps/reflect/src/components/auth/AuthContainer.tsx (hunk 2 above)
Comment (cancng) [reply apps/reflect/src/components/auth/AuthContainer.tsx]: I've replaced this component with the existing `AppSnackbar` component.

**Review by mertsabinov — CHANGES_REQUESTED** [review]
Please remove the unused imports `Alert` and `Snackbar`.

**Review by mertsabinov — APPROVED** [review]
LGTM

**Comment by batuhanbaybas** [comment]
Can you please set the checkbox to be top aligned instead of centered relative to the text? also the text has not correct align vertically maybe the width of the black area can be increased

**Comment by cancng** [comment]
> Can you please set the checkbox to be top aligned instead of centered relative to the text? also the text has not correct align vertically maybe the width of the black area can be increased also while user not select checkbox the auth buttons have to disable

I changed what you said but width of the black area remains same width

**Review by batuhanbaybas — COMMENTED** [review]
LGTM!

File: apps/reflect/src/components/auth/AuthLoginButton.tsx [hunk 3]
```diff
@@ -23,23 +23,25 @@ const AuthLoginButton: FC<IProps> = ({
...
               "&:hover": {
                 backgroundColor: "#202020",
                 borderRadius: "4px",
               },
             }
-          : {
-              cursor: "not-allowed",
-            }
-      }>
+          : { cursor: "not-allowed" }),
```
Comment (batuhanbaybas) [inline apps/reflect/src/components/auth/AuthLoginButton.tsx:35]: better the icon has grey color if disable state is true i think add grey color on stack fix that

Comment (batuhanbaybas) [reply apps/reflect/src/components/auth/AuthLoginButton.tsx:35]: also can you please add e ss for last update :)

Comment (cancng) [reply apps/reflect/src/components/auth/AuthLoginButton.tsx]: CleanShot 2023-05-20 at 23 41 37

File: apps/reflect/src/components/auth/AuthLoginButton.tsx [hunk 4]
```diff
@@ -32,16 +35,11 @@ const AuthLoginButton: FC<IProps> = ({
                 borderRadius: "4px",
               },
             }
-          : { cursor: "not-allowed" }),
+          : { cursor: "not-allowed", color: "gray" }),
       }}>
-      {icon}
+      {<Icon {...iconProps} {...(disabled && { color: "gray" })} />}
```
Comment (batuhanbaybas) [inline apps/reflect/src/components/auth/AuthLoginButton.tsx:40]: we realy need "..." on this props is it just take a value of true or false if i'm missing something please don't hesitate give a information 💯

**Review by batuhanbaybas — CHANGES_REQUESTED** [review]

File: apps/reflect/src/components/auth/AuthLoginButton.tsx (hunk 3 above)
Comment (batuhanbaybas) [reply apps/reflect/src/components/auth/AuthLoginButton.tsx]: yes 👍🏼

**Review by batuhanbaybas — APPROVED** [review]

File: apps/reflect/src/components/auth/AuthLoginButton.tsx (hunk 4 above)
Comment (cancng) [reply apps/reflect/src/components/auth/AuthLoginButton.tsx:40]: Yes we need it to add gray color when button is disabled. Also we need extra styles after on disabled styles

File: apps/reflect/src/components/auth/AuthLoginButton.tsx [hunk 5]
```diff
@@ -1,17 +1,20 @@
-import { FC, ReactNode } from "react";
+import { FC } from "react";
 import { Stack, Typography } from "@mui/material";
+import { IconBaseProps, IconType } from "react-icons";
 
 type IProps = {
   content: string;
-  icon: ReactNode;
+  icon: IconType;
+  iconProps?: IconBaseProps;
```
Comment (batuhanbaybas) [inline apps/reflect/src/components/auth/AuthLoginButton.tsx:8]: i think we are not need to use iconProps in here please give icon type of React node and send color parameter on perent component don't hesitate to ask if your mind is fixing

**Review by batuhanbaybas — APPROVED** [review]

**Review by mertsabinov — APPROVED** [review]

**Pull request merged**